packages/commonwealth/client/scripts/views/pages/CommunityManagement/CommunityProfile/CommunityProfileForm/CommunityProfileForm.tsx

```diff
@@ -0,0 +1,417 @@
+import { DefaultPage } from '@hicommonwealth/core';
+import { notifyError, notifySuccess } from 'controllers/app/notifications';
+import { featureFlags } from 'helpers/feature-flags';
+import getLinkType from 'helpers/linkType';
+import React, { useState } from 'react';
+import { slugifyPreserveDashes } from 'shared/utils';
+import app from 'state';
+import {
+  useEditCommunityBannerMutation,
+  useEditCommunityTagsMutation,
+} from 'state/api/communities';
+import { LinksArray, useLinksArray } from 'views/components/LinksArray';
+import {
+  CWCoverImageUploader,
+  ImageBehavior,
+} from 'views/components/component_kit/cw_cover_image_uploader';
+import { CWIcon } from 'views/components/component_kit/cw_icons/cw_icon';
+import { CWText } from 'views/components/component_kit/cw_text';
+import { CWTextArea } from 'views/components/component_kit/cw_text_area';
+import { CWForm } from 'views/components/component_kit/new_designs/CWForm';
+import { CWTag } from 'views/components/component_kit/new_designs/CWTag';
+import { CWTextInput } from 'views/components/component_kit/new_designs/CWTextInput';
+import { CWButton } from 'views/components/component_kit/new_designs/cw_button';
+import { CWRadioButton } from 'views/components/component_kit/new_designs/cw_radio_button';
+import { CWToggle } from 'views/components/component_kit/new_designs/cw_toggle';
+import { getCommunityTags } from '../../../manage_community/helpers';
+import './CommunityProfileForm.scss';
+import { CommunityTags, FormSubmitValues } from './types';
+import {
+  communityProfileValidationSchema,
+  linkValidationSchema,
+} from './validation';
+
+const CommunityProfileForm = () => {
+  const communityTagOptions: CommunityTags[] = ['DeFi', 'DAO'];
+  const community = app.config.chains.getById(app.activeChainId());
+
+  const [communityId] = useState(
+    slugifyPreserveDashes(community.id.toLowerCase()),
+  );
```

Comment:
do we really need this? you can use `community.id` down below, it is not gonna muate

Reply:
we can have it as const var too. I added state to memo the return value of `slugifyPreserveDashes`.